effekt/shared/src/main/scala/effekt/util/EditDistance.scala

```diff
@@ -0,0 +1,55 @@
+package effekt.util
+
+def editDistance(a: String, b: String, limit: Int): Option[Int] = {
+  val n = a.length
+  val m = b.length
+  if (n == 0) return Some(m)
+  if (m == 0) return Some(n)
+
+  // Early exit if minimum distance exceeds limit
+  val minDist = Math.abs(n - m)
+  if (minDist > limit) return None
+
+  // Strip common prefix
+  val prefixLen = a.view.zip(b.view).takeWhile { case (a, b) => a == b }.size
+
+  // Strip common suffix (but don't overlap with prefix)
+  val suffixLen = a.view.drop(prefixLen).reverse
+    .zip(b.view.drop(prefixLen).reverse)
+    .takeWhile { case (a, b) => a == b }
+    .size
+
+  val s1 = a.substring(prefixLen, n - suffixLen)
+  val s2 = b.substring(prefixLen, m - suffixLen)
+
+  // Levenshtein
+  val N = s1.length
+  val M = s2.length
+
+  // After stripping, if one is empty, distance is just the other's length
+  if (s1.isEmpty) return Some(M)
+  if (s2.isEmpty) return Some(N)
+
+  val d = Array.ofDim[Int](N + 1, M + 1)
+  for (i <- 0 to N) { d(i)(0) = i }
+  for (j <- 0 to M) { d(0)(j) = j }
```

Comment:
If this performs too badly, we can always use two or three rows of the array only... (it just makes it harder to read)